powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/task/CleaningTask.java

```diff
@@ -69,4 +72,32 @@ public void expireActivations() {
         activationServiceBehavior.expireActivations();
     }
 
+    @Scheduled(fixedRateString = "${powerauth.service.scheduled.job.retryFailedCallbackUrlEvent:3000}")
+    @SchedulerLock(
+            name = "retryFailedCallbackUrlEvent",
+            lockAtLeastFor = "#{T(java.lang.Math).round(${powerauth.service.scheduled.job.retryFailedCallbackUrlEvent:3000} * 0.8)}")
+    public void dispatchFailedCallbackUrlEvents() {
+        LockAssert.assertLocked();
+        logger.debug("dispatchFailedCallbackUrlEvents");
+        callbackUrlEventService.dispatchFailedCallbackUrlEvents();
+    }
+
+    @Scheduled(fixedRateString = "${powerauth.service.scheduled.job.dispatchPendingCallbackUrlEvent:3000}")
+    @SchedulerLock(
+            name = "dispatchPendingCallbackUrlEvent",
+            lockAtLeastFor = "#{T(java.lang.Math).round(${powerauth.service.scheduled.job.dispatchPendingCallbackUrlEvent:3000} * 0.8)}")
+    public void dispatchPendingCallbackUrlEvents() {
+        LockAssert.assertLocked();
+        logger.debug("dispatchPendingCallbackUrlEvents");
+        callbackUrlEventService.dispatchPendingCallbackUrlEvents();
+    }
+
```

Comment:
This is not an ideal solution as it brings another thread fighting for a PENDING events in the database. However, because of the skip locked queries, it should not cause an extensive blocking.